### DISCUSSION TITLE
Usability for digital designs?

### DISCUSSION THREAD

**Discussion by openpowerwtf** [gdsfactory/gdsfactory#765] [Q&A]
I have been looking at writing some Python code for KLayout for several uses in programmatically creating macros for digital designs using OpenROAD flow.  I can create a gds with gdsfactory and a lef from that with Magic and they do run through OpenROAD (but clearly some problems like wrong size and missing ports).

Any thoughts on whether gdsfactory is a good starting point for me?  Some ideas I have:

* blackbox macros for prototyping/analyzing: useful for floorplan/timing a new or current design in multiple technologies
* manually-placed standard cell macros for regular structures to optimize area/timing
* transistor-level custom cell design and macros for regular structures (memory/ALU/etc.)

**Comment by joamatab** [comment]
Hi Bill,

At the moment we only have a photonics flow, 

- we are starting to look into adding analog features such as spice #681 
- adding a OpenRoadFlow plugin for digital design would be great too. Once the open road developers release a python API it should be much easier to interface with gdsfactory

@jjcherry56
@maliberty
@eder-matheus

**Reply by openpowerwtf** [reply]
Thanks - I agree.  I've been doing some experimenting.  There should be some good uses in digital designs too.

**Comment by maliberty** [comment]
A python API is under development.  I'm unclear what is specifically needed here to satisfy the request.

**Comment by daquintero** [comment]
Already some microservice integration here for reference which will keep expanding between Openlane v1 and v2 
https://piel.readthedocs.io/en/latest/examples.html
